testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/admin/authentication/FlowTest.java

```diff
@@ -275,4 +275,150 @@ public void addExecutionFlow() {
+        flows = authMgmtResource.getFlows();
+        try {
+            Assert.assertEquals("New description", findFlowByAlias("New Flow", flows).getDescription());
+            assertAdminEvents.assertEvent(REALM_NAME, OperationType.UPDATE, AdminEventPaths.authEditFlowPath(found.getId()), ResourceType.AUTH_FLOW);
+        } finally {
+            response.close();
+        }
+
+        //Update name and description
+        found.setAlias("New Flow2");
+        found.setDescription("New description2");
+        response = authMgmtResource.updateFlow(found.getId(), found);
+        flows = authMgmtResource.getFlows();
+        try {
+            Assert.assertEquals("New Flow2", findFlowByAlias("New Flow2", flows).getAlias());
+            Assert.assertEquals("New description2", findFlowByAlias("New Flow2", flows).getDescription());
+            assertAdminEvents.assertEvent(REALM_NAME, OperationType.UPDATE, AdminEventPaths.authEditFlowPath(found.getId()), ResourceType.AUTH_FLOW);
+            Assert.assertNull(findFlowByAlias("New Flow", flows));
+        } finally {
+            response.close();
+        }
+        authMgmtResource.deleteFlow(testFlow.getId());
+    }
+
+    @Test
+    public void editExecutionFlowTest() {
+        HashMap<String, String> params = new HashMap<>();
+        List<AuthenticationExecutionInfoRepresentation> executionReps;
+        //create new parent flow
+        AuthenticationFlowRepresentation newFlow = newFlow("Parent-Flow", "This is a parent flow", "basic-flow", true, false);
+        createFlow(newFlow);
+
+        //create a child sub flow
+        params.put("alias", "Child-Flow");
+        params.put("description", "This is a child flow");
+        params.put("provider", "registration-page-form");
+        params.put("type", "basic-flow");
+
+        authMgmtResource.addExecutionFlow("Parent-Flow", params);
+        assertAdminEvents.assertEvent(REALM_NAME, OperationType.CREATE, AdminEventPaths.authAddExecutionFlowPath("Parent-Flow"), params, ResourceType.AUTH_EXECUTION_FLOW);
+
+        //create another
+        executionReps = authMgmtResource.getExecutions("Parent-Flow");
+
+        params = new HashMap<>();
+        params.put("alias", "Child-Flow");
+        params.put("description", "This is another child flow");
+        params.put("provider", "registration-page-form");
+        params.put("type", "basic-flow");
+
+        try {
+            authMgmtResource.addExecutionFlow("Parent-Flow", params);
+            Assert.fail("addExecutionFlow the alias already exist");
+        } catch (Exception expected) {
+            // Expected
+        }
+
+        AuthenticationExecutionInfoRepresentation found = executionReps.get(0);
+        found.setDisplayName("Parent-Flow");
+
+        Response response = authMgmtResource.updateExecutions("Parent-Flow", found);
+        try {
+            Assert.assertEquals("updateFlow using the alias of existing flow should fail", 409, response.getStatus());
```

Comment:
This will need to catch exception instead of checking the response status. Which is in general better approach.

Reply:
Edited